woocommerce/woocommerce-gutenberg-products-block
# Add/block product categories

Fixes #212  – Add a new block for a Product Categories list. This is using another approach to rendering, using the JS on the front-end as well. I've put up this draft PR to get feedback on this as an approach, since I think we'll want to do something similar for the 2 product reviews blocks (#588 & #589).

Currently you can add the block, and it renders on the frontend, but you can't reload or re-open the editor with an existing Product Categories List block– something's breaking the attribute validation.

**Questions**

- Does the folder & component structure make sense?
- Should we attempt to render any content into the `save` function? (loading, some skeleton, ?)
- Thoughts on this approach in general?

For now, just cc'ing @woocommerce/blocks – will request proper reviews once this is ready.

Still to do:

- Fix whatever bug is preventing loading an existing block
- Add links to the links
- Add navigation code to the dropdown in an accessible way
- Make sure nav only happens on frontend
- Add hierarchy option to dropdown view

#### Accessibility

Before un-drafting I'll update this section.


### Screenshots

![Screen Shot 2019-05-30 at 5 37 20 PM](https://user-images.githubusercontent.com/541093/58666567-afd9c000-8301-11e9-9d40-d94d63bee27d.png)
![Screen Shot 2019-05-30 at 5 37 09 PM](https://user-images.githubusercontent.com/541093/58666568-b0725680-8301-11e9-944f-a7fa4bac67dd.png)
![Screen Shot 2019-05-30 at 5 36 53 PM](https://user-images.githubusercontent.com/541093/58666569-b0725680-8301-11e9-96ed-889f6eac2bb6.png)

### How to test the changes in this Pull Request:

1. Add a Product Categories List block
2. You should be able to toggle all the toggles, and have that reflect on the preview and in the frontend.
